kong/runloop/balancer/init.lua

```diff
@@ -472,7 +473,8 @@ local function set_host_header(balancer_data, upstream_scheme, upstream_host, is
 
     var.upstream_host = new_upstream_host
 
-    if is_balancer_phase then
+    -- stream module does not support ngx.balancer.recreate_request
+    if is_balancer_phase and is_http_module then
```

Comment:
Talked offline with @oowl. He says the scenario using this feature is not affected, and the fix for `recreate_request` will be written later.

Reply:
I will do a patch for Openresty after 3.4 released.

Reply:
I have read nginx source code, nginx will compile all proxy var when init upstream connection https://github.com/nginx/nginx/blob/master/src/stream/ngx_stream_proxy_module.c#L1070